resource-managers/yarn/src/test/scala/org/apache/spark/deploy/yarn/ClientSuite.scala

```diff
@@ -685,9 +685,9 @@ class ClientSuite extends SparkFunSuite with Matchers {
     ("files URI unmatch test1", "file:///file1", "file://host/file2"),
     ("files URI unmatch test2", "file://host/file1", "file:///file2"),
     ("files URI unmatch test3", "file://host/file1", "file://host2/file2"),
-    ("wasb URI unmatch test1", "wasb://bucket1@user", "wasb://bucket2@user/"),
-    ("wasb URI unmatch test2", "wasb://bucket1@user", "wasb://bucket1@user2/"),
-    ("s3 URI unmatch test", "s3a://user@pass:bucket1/", "s3a://user2@pass2:bucket1/"),
+    ("wasb URI unmatch test1", "wasb://user@bucket1", "wasb://user@bucket2/"),
+    ("wasb URI unmatch test2", "wasb://user@bucket1", "wasb://user2@bucket1/"),
+    ("s3 URI unmatch test", "s3a://user:pass@bucket1/", "s3a://user2:pass2@bucket1/"),
     ("hdfs URI unmatch test1", "hdfs://namenode1/path1", "hdfs://namenode1:8080/path2"),
     ("hdfs URI unmatch test2", "hdfs://namenode1:8020/path1", "hdfs://namenode1:8080/path2")
```

Comment:
Hi, @zekai-li . There is no additional test coverage for your reported case. Do you think you can add one to prevent the future regression?

Reply:
I added a test data for match：hdfs://localhost:8080 hdfs://127.0.0.1:8080

Reply:
Thanks, @zekai-li . Does it fail before this patch?

Reply:
It was a failure. As described by my pr. It's too sketchy to tell by uri.getauthority